specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/managedDatabases.json

```diff
@@ -0,0 +1,556 @@
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the database.",
+            "schema": {
+              "$ref": "#/definitions/ManagedDatabase"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ResourceNotFound - The requested resource was not found."
+          },
+          "202": {
+            "description": "Accepted"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Updates a managed database with minimal properties": {
+            "$ref": "./examples/ManagedDatabaseUpdateMin.json"
+          },
+          "Updates a managed database with maximal properties": {
+            "$ref": "./examples/ManagedDatabaseUpdateMax.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ManagedDatabaseProperties": {
+      "description": "The managed database's properties.",
+      "type": "object",
+      "properties": {
+        "collation": {
+          "description": "Collation of the managed database.",
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "status": {
+          "description": "Status for the database.",
+          "enum": [
+            "Online",
+            "Offline",
+            "Shutdown",
+            "Creating",
+            "Inaccessible"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ManagedDatabaseStatus",
+            "modelAsString": true
+          }
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "Creation date of the database.",
+          "type": "string",
+          "readOnly": true
+        },
+        "pointInTime": {
+          "format": "date-time",
+          "description": "Conditional. If createMode is PointInTimeRestore, this value is required. Specifies the point in time (ISO8601 format) of the source database that will be restored to create the new database.",
+          "type": "string",
+          "readOnly": true
+        },
+        "defaultSecondaryLocation": {
+          "description": "Geo pared region.",
+          "type": "string",
+          "readOnly": true
+        },
+        "catalogCollation": {
+          "description": "Collation of the metadata catalog.",
+          "enum": [
+            "DATABASE_DEFAULT",
+            "SQL_Latin1_General_CP1_CI_AS"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "CatalogCollationType",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "createMode": {
```

Comment:
Have you considered using a [discriminator](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-discriminator-value) ?

Reply:
Could you please point me to an example in code so I can see what I need to do in order for this discriminator to be generated from my code? I would like to have my code inline with generated swagger, or is it possible to modify the swagger alone since it doesn't have anything to do with the logic?

Reply:
It is about modifying the swagger.
[Postgresql](https://github.com/Azure/azure-rest-api-specs/blob/213f67c53ae4f46ced50564f5e4b34d245f2cd9e/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json#L1223) swagger is a good example for the discriminator extensions.